app/models/course/material.rb

```diff
@@ -28,7 +30,11 @@ def record_userstamp
 
   def initialize_duplicate(duplicator, other)
     self.attachment = duplicator.duplicate(other.attachment)
-    self.folder = duplicator.duplicate(other.folder)
+    self.folder = if duplicator.duplicated?(other.folder)
```

Comment:
If I'm not wrong, you are relying on an implicit assumption that all concrete folders are to be duplicated before the materials are duplicated. I would rather that not be the case, is there a way to do this differently? Perhaps specifying the behavior more explicitly between `:course` and `:object` modes?